services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/kafka/KafkaBasedMappingAndDelegatingCommandHandler.java

```diff
@@ -42,13 +49,25 @@
  */
 public class KafkaBasedMappingAndDelegatingCommandHandler extends AbstractMappingAndDelegatingCommandHandler {
 
+    /**
+     * Duration after which record processing is cancelled.
+     * <p>
+     * Usually, the timeout values configured for the service clients involved in command processing should define the
+     * overall time limit. The timeout mechanism associated with this value here serves as a safeguard against
+     * exceptional cases where the other timeouts didn't get applied. It thereby prevents command processing from
+     * getting stuck, so that the command consumer doesn't get blocked.
+     */
+    private static final Duration PROCESSING_TIMEOUT = Duration.ofMinutes(1);
```

Comment:
one minute? what's the reasoning behind such a long timeout? I would have expected something in the single digit second range ...

Reply:
The value here shall serve as an upper boundary of configured timeout values related to command processing. Therefore I've chosen a high value. (And in the Kafka world, timeouts in general seem to be quite high by default, e.g. producer.send() with 2min timeout. So, that was an influence here.)

But right, this may be unnecessarily high (also in connection with #2888, reducing producer default timeouts).
How about 8s?

Reply:
Sounds a lot better to me :-) However, the issue indicates that this value should be configurable or am I mistaken?

Reply:
It would be best to have the value configurable, yes. However, I would wait for #2904 here.
In any case, configuration of the other timeouts (device registry, infinispan clients) is more important. The timeout above should only trigger in exceptional cases (like the Hot Rot client NPE mentioned in #2919).

Reply:
ok